src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,138 @@
+import { useEffect, useState } from 'react';
+
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+// eslint-disable-next-line import/extensions
+import { PeoplePageTitle } from '../page/PeoplePageTitle';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
+import { QueryParams } from '../enum/queryParams.enum';
+import { Sex } from '../enum/sex.enum';
 
 export const PeoplePage = () => {
+  // #region states
+  const [loading, setLoading] = useState(true);
+  const [fetchPeopleError, setFetchPeopleError] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [query, setQuery] = useState('');
+  const [showFilters, setShowFilters] = useState(false);
+  // #endregion
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const activeCenturies = searchParams.getAll(QueryParams.Centuries) || [];
+  const activeSex = searchParams.get(QueryParams.Sex) as Sex || '';
+  const activeQuery = searchParams.get(QueryParams.Query) || '';
+
+  const toggleCenturies = (currentCentury: number) => {
+    const tempCenturies = !activeCenturies.includes(String(currentCentury))
+      ? [...activeCenturies, `${currentCentury}`]
+      : activeCenturies.filter(century => century !== String(currentCentury));
+
+    return tempCenturies;
+  };
+
+  const normalizedQuery = query.trim().toLowerCase();
+  const filteredPeople = people.filter(person =>
+    person.name.toLowerCase().includes(normalizedQuery),
+  );
+
+  const filterByCentury = () => {
+    if (activeCenturies.length) {
+      return filteredPeople.filter(person =>
+        activeCenturies.includes(String(Math.floor(person.born / 100) + 1)),
+      );
+    } else {
+      return filteredPeople;
+    }
+  };
+
+  const handleFilterBySex = (peopleList: Person[], activeSex: Sex) => {
```

Comment:
'activeSex' is already declared in the upper scope on line 26 column 9.
If u need this variable, do not pass it to the function, or change the name to avoid confusion